cachi2/core/package_managers/cargo/main.py

```diff
@@ -1,17 +1,53 @@
 import logging
 import textwrap
+from dataclasses import dataclass
+from functools import cached_property
+from itertools import chain
+from pathlib import Path
+from typing import Optional
 
 import tomlkit
+from packageurl import PackageURL
 from tomlkit.toml_file import TOMLFile
 
+from cachi2.core.errors import PackageRejected
 from cachi2.core.models.input import Request
 from cachi2.core.models.output import Component, EnvironmentVariable, ProjectFile, RequestOutput
 from cachi2.core.rooted_path import RootedPath
+from cachi2.core.scm import get_repo_id
 from cachi2.core.utils import run_cmd
 
 log = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True)
+class CargoPackage:
+    """CargoPackage."""
+
+    name: str
+    version: str
+    source: Optional[str] = None  # [git|registry]+https://github.com/<org>/<package>#[|<sha>]
+    checksum: Optional[str] = None
+    dependencies: Optional[list] = None
+    vcs_url: Optional[str] = None
+
+    @cached_property
+    def purl(self) -> PackageURL:
```

Comment:
Non-blocking question: Do we want/need to handle alternate registries? If so, I would envision that as a follow-up

Reply:
Alternate registries are theoretically possible, but to the best of my knowledge are not widely used. I think we can safely postpone adding their support.